plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -118,15 +126,22 @@ private static Metrics computeMetrics(Schema icebergSchema, ColumnMetadata<OrcTy
             ColumnStatistics orcColumnStats = columnStatistics.get().get(orcColumnId);
             int icebergId = getIcebergId(orcColumn);
             Types.NestedField icebergField = icebergSchema.findField(icebergId);
+            MetricsModes.MetricsMode metricsMode = MetricsUtil.metricsMode(icebergSchema, metricsConfig, icebergId);
+            if (metricsMode == MetricsModes.None.get()) {
+                continue;
+            }
             verify(icebergField != null, "Cannot find Iceberg column with ID %s in schema %s", icebergId, icebergSchema);
             valueCountsBuilder.put(icebergId, fileRowCount);
             if (orcColumnStats.hasNumberOfValues()) {
                 nullCountsBuilder.put(icebergId, fileRowCount - orcColumnStats.getNumberOfValues());
             }
-            toIcebergMinMax(orcColumnStats, icebergField.type()).ifPresent(minMax -> {
-                lowerBoundsBuilder.put(icebergId, minMax.getMin());
-                upperBoundsBuilder.put(icebergId, minMax.getMax());
-            });
+
+            if (metricsMode != MetricsModes.Counts.get()) {
```

Comment:
`MetricsMode` is an interface, open for extensions. The fact that something isn't Counts doesn't mean it's Full.

Also, there is no javadoc indicating `.get()` returns same instance every time, so using `.equals()` would be more appropriate here.

this should be handled in a defensive manner:

```
if (metricsMode.equals(MetricsModes.None.get())) {
 // nothing to do
}
else if (metricsMode.equals(MetricsModes.Counts.get()) {
 .. collect counts 
}
else if (metricsMode.equals( full )) {
  ...
}
else {
    throw new UnsupportedOperationException("Unsupported metrics mode: " + metricsMode);
}
```

Reply:
@findepi The conditional statement here handles the case where the metricsMode is truncate(length) or full. Both of these two cases need to store min/max except that truncate need to apply truncation if needed. 

All of the metrics modes other than None need to collect counts. There would be duplication of code if we are to handle each mode separately. Can we add a `verify` function instead to make sure that the metric mode is supported if we want to handle it more defensively?